server/pulp/server/async/scheduler.py

```diff
@@ -317,15 +320,60 @@ def tick(self):
         :return:    number of seconds before the next tick should run
         :rtype:     float
         """
-        ret = super(Scheduler, self).tick()
         self._failure_watcher.trim()
-
         # this is not an event that gets sent anywhere. We process it
         # immediately.
         scheduler_event = {'timestamp': time.time(),
                            'type': 'scheduler-event',
-                           'hostname': ("%s@%s" % (SCHEDULER_WORKER_NAME, platform.node()))}
+                           'hostname': ("%s@%s@%s" % (SCHEDULER_WORKER_NAME,
+                                                      str(celery_name), platform.node()))}
         worker_watcher.handle_worker_heartbeat(scheduler_event)
+
+        celerybeat_name = SCHEDULER_WORKER_NAME + "@" + str(celery_name) + "@" + platform.node()
+
+        # Establishing connection with DB
+        db = db_connection.get_database()
+        collection = db['celerybeat_lock']
+
+        # Creating a unique constraint on the collection
+        collection.create_index('lock', unique=True)
+
+        _logger.debug('Updating the lock with celerybeat_name %s ', celerybeat_name)
```

Comment:
This should be internationalized with _() and you need to switch the logging to dictionary based (not position based).